### PR TITLE
Manual Provider Space Support

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

Implements `environs.NetworkingEnviron` for the manual provider and indicates that spaces are supported.

## QA steps

I did this via VMs that I provisioned with MAAS.

- Provision a machine and bootstrap onto it via `juju bootstrap manual/ubuntu@<IP> spaces --no-gui --debug --build-agent`.
- Provision 2 more machines each with a NIC in a different subnet via `juju add-machine ssh:ubuntu@<IP>`.
- `juju spaces` should list the 2 subnets in the alpha space.
- `juju add-space space-<ID> <CIDR>` for the ID/subnet of machines 0 and 1 respectively.
- `juju model-config container-networking-method=provider`.
- `juju deploy postgresql --to lxd:0 --bind space-0` should set up a bridge on machine 0 and deploy to LXD successfully.
- `juju deploy percona-cluster --to lxd:0 --bind space-1` will fail to start a container because machine 0 does not have a NIC in the space.
- Retrying the deployment to lxd:1 should succeed.

## Documentation changes

Will require documentation of manual provider space support.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Enable Spaces for Manual Provider](https://trello.com/c/4Iglhzz2/2844-enable-spaces-for-manual-provider)

File: provider/manual/environ_test.go [hunk 1]
```diff
@@ -147,11 +147,6 @@ exit 0
 	}
 }
 
-func (s *environSuite) TestSupportsNetworking(c *gc.C) {
-	_, ok := environs.SupportsNetworking(s.env)
-	c.Assert(ok, jc.IsFalse)
```
Comment (SimonRichardson) [inline provider/manual/environ_test.go:152]: Rather than get rid of the test, can we just do `js.IsTrue`.

Comment (manadart) [reply provider/manual/environ_test.go:152]: It's represented in the new `environNetworkSuite`.

**Review by SimonRichardson — APPROVED** [review]
I tested with LXD using the manual integration tests. Obviously I only had one nic, but worked fine.

File: provider/manual/environ_test.go (hunk 1 above)
Comment (SimonRichardson) [reply provider/manual/environ_test.go:152]: ah

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**